### PR TITLE
[Android] Added ref code for installs from ads

### PR DESCRIPTION
**Pull request by samartnik**
Resolves https://github.com/brave/brave-browser/issues/25265

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by porteron — APPROVED** [review]
looks good!

File: android/java/org/chromium/chrome/browser/util/BraveReferrer.java [hunk 1]
```diff
@@ -36,6 +36,7 @@ public class BraveReferrer implements InstallReferrerStateListener {
     private static final String APP_CHROME_DIR = "app_chrome";
     private static final String PROMO_CODE_FILE_NAME = "promoCode";
     private static final String BRAVE_REFERRER_RECEIVED = "brave_referrer_received";
+    private static final String GOOGLE_ADS_REFERRAL_CODE = "UAC001";
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/util/BraveReferrer.java:39]: how did we end up with that particular code?

Comment (samartnik) [reply android/java/org/chromium/chrome/browser/util/BraveReferrer.java:39]: It was suggested by @porteron. It's our own code so it doesn't matter what exactly we choose as long as it is used on both client and server side.

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Samsung S10+` running `Android 12` using the following build(s):

```
Brave | 1.46.32 Chromium: 106.0.5249.91 (Official Build) canary (64-bit)
--- | ---
Revision | fa96d5f07b1177d1bf5009f647a5b8c629762157-refs/branch-heads/5249@{#707}
OS | Android 12; Build/SP1A.210812.016
```

**`Test Case #1`**

As per https://github.com/brave/brave-browser/issues/25265#issuecomment-1267633910, we'll need to run through this case once we get https://github.com/brave/brave-core/pull/15321 merged into `1.45.x` and new builds have been created. Discussed with @porteron and we should be fine if `Test Case #1` & `Test Case #2` work on `Nightly` and `Release`. As long as `Test Case #1` works on `BETA`, we're golden 👍 

**`Test Case #2`**

* downloaded/installed `Brave Nightly` from GPS
* launched Brave and ensured that `"promo_code": "BRV001"` via `brave://local-state`

```
"referral": {
         "initialization": true,
         "promo_code": "BRV001"
```

**`Example`** | **`Example`**
----------------|----------------
Screenshot_20221004-233812_Google Play Store | Screenshot_20221004-233911_Brave - Nightly

**`Test Case #3`**

* opened https://kjozwiak.github.io via `Chrome` and clicked on `Nightly GPS TVT141`
* downloaded/installed `Brave Nightly` from GPS
* launched Brave and ensured that `"promo_code": "TVT141"` via `brave://local-state`

```
"referral": {
         "download_id": "ce4a57ab-bae8-40af-911f-9b736e5b2290",
         "initialization": true,
         "promo_code": "TVT141"
```

**`Example`** | **`Example`**
----------------|----------------
Screenshot_20221004-234438_Google Play Store | Screenshot_20221004-234532_Brave - Nightly

**Pull request merged**